src/api/app/helpers/webui/notification_helper.rb

```diff
@@ -28,6 +28,28 @@ def request_badge_color(state)
     end
   end
 
+  def action_description(notification)
+    case notification.event_type
+    when 'Event::RequestStatechange', 'Event::RequestCreate', 'Event::ReviewWanted', 'Event::CommentForRequest'
+      source_and_target(notification)
+    when 'Event::CommentForProject'
+      "#{notification.notifiable.commentable.name}"
+    when 'Event::CommentForPackage'
+      commentable = notification.notifiable.commentable
+      "#{commentable.project.name} / #{commentable.name}"
+    end
+  end
+
+  def source_and_target(notification)
+    capture do
+      if notification.source.present?
```

Comment:
Hmm, this only work with a NotificationPresenter object right? Why did you moved all the logic to the helpers?

Reply:
Because I needed to add the arrow icon in the middle. But I'm working on dropping `source` and `target` from the presenter and just have all this logic in the helper. But it is not a blocker, it's something I can in another PR.

Reply:
ok